freqtrade/strategy/interface.py

```diff
@@ -707,35 +721,53 @@ def _analyze_ticker_internal(self, dataframe: DataFrame, metadata: dict) -> Data
         # always run if process_only_new_candles is set to false
         if (not self.process_only_new_candles or
                 self._last_candle_seen_per_pair.get(pair, None) != dataframe.iloc[-1]['date']):
+
+            populate_indicators = not external_data
             # Defs that only make change on new candle data.
-            dataframe = self.analyze_ticker(dataframe, metadata)
+            dataframe = self.analyze_ticker(dataframe, metadata, populate_indicators)
+
             self._last_candle_seen_per_pair[pair] = dataframe.iloc[-1]['date']
-            self.dp._set_cached_df(
-                pair, self.timeframe, dataframe,
-                candle_type=self.config.get('candle_type_def', CandleType.SPOT))
+
+            candle_type = self.config.get('candle_type_def', CandleType.SPOT)
+            self.dp._set_cached_df(pair, self.timeframe, dataframe, candle_type=candle_type)
+
+            if not external_data:
+                self.rpc.send_msg(
+                    {
+                        'type': RPCMessageType.ANALYZED_DF,
+                        'data': {
+                            'key': (pair, self.timeframe, candle_type),
+                            'value': dataframe_to_json(dataframe)
+                        }
+                    }
+                )
+
```

Comment:
I'd move this to the dataprovider.
rpc should not be exposed to the strategy directly (this makes it accessible via `self.rpc` from user code, too) to avoid undebuggable problems.

you can then pass rpc to the dataprovider as optional __init__ argument (please store it in a "really" private variable (`__rpc`)).

You'll need to be aware that `_set_cached_df()` is also used in backtesting - so if you do additional calls there - it must default to false.
I'd probably add a new private method for that though - `_set_cached_df()` should do what it's supposed to, without surprising side-effects.
An explicit call to "if not external: self.dp.emit_df()" might show the intend in a clearer way.

Reply:
:+1: